semver/parse.ts

```diff
@@ -41,19 +41,19 @@ export function parse(version: string | SemVer): SemVer {
   if (!groups) throw new TypeError(`Invalid Version: ${version}`);
 
   // these are actually numbers
-  const major = parseInt(groups.major);
-  const minor = parseInt(groups.minor);
-  const patch = parseInt(groups.patch);
+  const major = parseInt(groups.major!);
+  const minor = parseInt(groups.minor!);
+  const patch = parseInt(groups.patch!);
```

Comment:
Ignoring the lint error with `!` as `parseInt(undefined)` will parse to `NaN` and I've added a new isNaN check below.